packages/genesys-spark-components/src/components/stable/gux-time-picker/gux-time-picker.service.ts

```diff
@@ -43,12 +44,15 @@ export function getLocaleClockType(root: HTMLElement): GuxClockType {
     timeZone: 'UTC'
   }).format(date);
 
-  // The localization team has requested that some locales be hardcoded to the 24h clock. https://inindca.atlassian.net/browse/LOCAL-9597
-  const localesSetTo24h = ['ar', 'ko', 'zh-cn', 'zh-tw'];
-  if (
-    localesSetTo24h.some(localeSetTo24h => locale.startsWith(localeSetTo24h))
-  ) {
-    return '24h';
+  // LEGACY -- The localization team has requested that some locales be hardcoded to the 24h clock. https://inindca.atlassian.net/browse/LOCAL-9597
+  // Localization team has requested to removing hardcoded 24h clock locales as requested in https://inindca.atlassian.net/browse/COMUI-2890
+  if (!useRegionalDates()) {
```

Comment:
Just for my own understanding, so if  `useRegionalDates` is not being used then those locales should be set to `24hr` and if not they are set to `12hr`